### PR TITLE
dojot/dojot#2380 | dojot/dojot#2381 - Add certificates filter using k…

### PR DESCRIPTION
**Pull request by heitorlucas2018**
…ey Value

* **Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)
  Do not filter how much nested attribute

* **What is the new behavior (if this is a feature change)?**
  Filter using any attribute passed by the keyVal query parameter

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: iam/x509-identity-mgmt/js/src/db/CommonModel.js [hunk 1]
```diff
@@ -31,6 +31,20 @@ class CommonModel {
   sanitizeFields(cert) {
     return this.mongoClient.sanitizeFields(cert, this.projectableFields);
   }
+
+  handleFilterField(candidates) {
+    if (!candidates) {
+      return {};
+    }
+
+    const filterField = candidates.split(',').reduce((e, i) => {
```
Comment (rodrigoaasm) [inline iam/x509-identity-mgmt/js/src/db/CommonModel.js:40]: Por que foi utilizado um reducer ao invés de um filter ou um mapper ?

**Review by rodrigoaasm — COMMENTED** [review]
I am leaving a question. The rest is fine with me.

**Review by luissajunior — APPROVED** [review]

File: iam/x509-identity-mgmt/js/src/db/CommonModel.js (hunk 1 above)
Comment (heitorlucas2018) [reply iam/x509-identity-mgmt/js/src/db/CommonModel.js:40]: para converter um array em object json

**Review by rodrigoaasm — APPROVED** [review]
It's ok for me now. Good job.

**Pull request merged**